### PR TITLE
Catch and log all exception in CosmosDBMetricsProvider, 4.8.1 version upgrade

### PR DESCRIPTION
**Pull request by chiangvincent**
We should be catching and logging all exceptions thrown during the CosmosDBMetricsProvider

### CODE REVIEW

File: src/WebJobs.Extensions.CosmosDB/Trigger/CosmosDBMetricsProvider.cs [hunk 1]
```diff
@@ -96,6 +96,10 @@ public async Task<CosmosDBTriggerMetrics> GetMetricsAsync()
 
                 _logger.LogWarning(Events.OnScaling, errormsg);
             }
+            catch (Exception e)
+            {
+                _logger.LogWarning(Events.OnScaling, "Exception occurred while obtaining metrics for CosmosDB {0}: {1}.", e.GetType().ToString(), e.Message);
```
Comment (ealsur) [inline src/WebJobs.Extensions.CosmosDB/Trigger/CosmosDBMetricsProvider.cs]: Log the exception as is, not just the message?

Comment (ealsur) [reply src/WebJobs.Extensions.CosmosDB/Trigger/CosmosDBMetricsProvider.cs]:
```suggestion
                _logger.LogWarning(Events.OnScaling, "Exception occurred while obtaining metrics for CosmosDB {0}: {1}.", e.GetType().ToString(), e);
```

Comment (ealsur) [reply src/WebJobs.Extensions.CosmosDB/Trigger/CosmosDBMetricsProvider.cs]: Given it's an unknown you probably want to log as much as you can

**Review by ealsur — APPROVED** [review]

File: src/WebJobs.Extensions.CosmosDB/Trigger/CosmosDBMetricsProvider.cs (hunk 1 above)
Comment (chiangvincent) [reply src/WebJobs.Extensions.CosmosDB/Trigger/CosmosDBMetricsProvider.cs]: changed to log e.tostring()!

File: build/common.props [hunk 2]
```diff
@@ -3,7 +3,7 @@
     <!-- Extensions can have independent versions and only increment when released -->
     <Version>3.0.0$(VersionSuffix)</Version>
     <ExtensionsVersion>5.0.0$(VersionSuffix)</ExtensionsVersion> <!-- WebJobs.Extensions -->
-    <CosmosDBVersion>4.8.0$(VersionSuffix)</CosmosDBVersion>
+    <CosmosDBVersion>4.9.0$(VersionSuffix)</CosmosDBVersion>
```
Comment (ealsur) [inline build/common.props]:
```suggestion
    <CosmosDBVersion>4.8.1$(VersionSuffix)</CosmosDBVersion>
```

Use a patch version. Minor version is used when dependencies or APIs change

**Pull request merged**